dali/python/backend_impl.cc

```diff
@@ -870,6 +896,19 @@ void ExposeTensorList(py::module &m) {
         }),
       "tl"_a,
       "layout"_a = py::none())
+    .def(py::init([](py::list &list_of_tensors, string layout = "") {
```

Comment:
While testing this constructor I experienced that when it was declared after the one from `py::object` the later was called when constructing TL with `TensorListGPU([TensorGPU])` and when I moved it up the problem was resolved. Is this ambiguity something we should be concerned about, especially considering that there is already a similar ambiguity with the `py::capsule` constructor?

Reply:
It's how pybind operates, it will try to match the constructors in succession, so the narrowest-matching ones should be higher than the more generic ones.